Composer/plugins/azurePublish/src/node/azureResourceManager/azureResourceManager.ts

```diff
@@ -203,6 +205,214 @@ export class AzureResourceMananger {
     }
   }
 
+  /**
+   * QnA Resource depends on serveral components, including appinights and web config
+   * @param config 
+   */
+  public async deployQnAReource(config: QnAResourceConfig): Promise<{ endpoint: string; subscriptionKey: string }> {
+    try {
+      this.logger({
+        status: BotProjectDeployLoggerType.PROVISION_INFO,
+        message: 'Deploying QnA Resource ...',
+      });
+
+      // initialize the name
+      const qnaMakerSearchName = `${config.accountName}-search`.toLowerCase().replace('_', '');
+      const qnaMakerWebAppName = `${config.accountName}-qnahost`.toLowerCase().replace('_', '');
+      const qnaMakerServiceName = `${config.accountName}-qna`;
+      // deploy search service
+      const searchManagementClient = new SearchManagementClient(this.creds, this.subscriptionId);
+      const searchServiceDeployResult = await searchManagementClient.services.createOrUpdate(config.resourceGroupName, qnaMakerSearchName, {
+        location: config.location,
+        sku: {
+          name: 'standard'
+        },
+        replicaCount: 1,
+        partitionCount: 1,
+        hostingMode: 'default'
+      });
+
+      if (searchServiceDeployResult._response.status >= 300) {
+        this.logger({
+          status: BotProjectDeployLoggerType.PROVISION_ERROR,
+          message: searchServiceDeployResult._response.bodyAsText,
+        });
+        throw new Error(searchServiceDeployResult._response.bodyAsText);
+      }
+
+      // deploy websites
+      // Create new Service Plan or update the exisiting service plan created before
+      const webSiteManagementClient = new WebSiteManagementClient(this.creds, this.subscriptionId);
+      const servicePlanName = config.resourceGroupName;
+      const servicePlanResult = await webSiteManagementClient.appServicePlans.createOrUpdate(
+        config.resourceGroupName,
+        servicePlanName,
+        {
+          location: config.location,
+          sku: {
+            name: 'S1',
+            tier: 'Standard',
+            size: 'S1',
+            family: 'S',
+            capacity: 1,
+          },
+        }
+      );
+
+      if (servicePlanResult._response.status >= 300) {
+        this.logger({
+          status: BotProjectDeployLoggerType.PROVISION_ERROR,
+          message: servicePlanResult._response.bodyAsText,
+        });
+        throw new Error(servicePlanResult._response.bodyAsText);
+      }
+
+      // deploy or update exisiting app insights component
```

Comment:
Same as above -- since we are already doing app insights configurations, I don't see why we need to do it again here.  

I would prefer that we do it ONCE, and then do the necessary additional configuration in the same manner as it is being done with the web app -- that is, IF app insights AND qna are both enabled, the appropriate configuration keys are configured in a separate step.

Reply:
Or perhaps should this be a DIFFERENT app insights resource that has a unique name instead of sharing the same resource with the web app?

Reply:
> Or perhaps should this be a DIFFERENT app insights resource that has a unique name instead of sharing the same resource with the web app?

If we already have a application insights with the same name under the same resource group, it won't be created again, this is handled by the azure api (the method is createOrUpdate). But since the application insights is a necessary dependency of qna maker, we don't know whether an app insights instance has been created before, so I create or update any existing app insights under that resource group to make sure there exists an instance.

Do we need to split the app insights to two different resources (one is for webapp, one is for qna)?